### ISSUE TITLE
[css-cascade-6] What is the desired shape of CSSScopeRule?

### ISSUE THREAD

**Issue by mirisuzanne** [w3c/csswg-drafts#8626]
In https://github.com/w3c/csswg-drafts/pull/8468 @andruud pushed a spec definition for adding `CSSScopeRule` to CSSOM:

```
	[Exposed=Window]
	interface CSSScopeRule : CSSGroupingRule {
	  readonly attribute CSSOMString start;
	  readonly attribute CSSOMString end;
	};
```

As far as I can tell, this change in the spec never had a resolution attached – so bringing it up here to get that resolution. In conversation with @fantasai, she also raised a few questions about the final shape of this rule:

* Is 'start'/'end' the terms we want? Or do we want 'root'/'floor', or any number of other things. Since this is the only place in which the terminology is actually user-facing, it becomes a real question.
* Should they return null string or null if they're missing?
* Do they include the wrapping parentheses or not?

**Comment by fantasai** [comment]
Fwiw, I think we should not include the wrapping parentheses so that the returned string is a proper selector, and they should be null if omitted.

I don't have much opinion on the names (they seem fine), but I think it's a question worth asking.

**Comment by mirisuzanne** [comment]
I agree that the names seem fine to me. Other terms that get used for these concepts are 'root', and 'boundary' or 'limit'.

**Comment by tabatkins** [comment]
I'm fine with the names as well. I could think of some different ones, but nothing seems *better*, and quite a bit seem worse. 

`null` when omitted is indeed correct imo. There's a lot of stuff in CSSOM that uses empty string for this semantic and I think that's just terrible legacy design, and doesn't need to be emulated.

Yeah, omit the wrapping parens so the values are just the selectors themselves. The parens are a syntactic necessity for the rule itself, they're not part of the start/end bits.

**Comment by tabatkins** [comment]
Yeah that's a good point, so let's stick with the more generic `start` and `end`.

**Comment by mirisuzanne** [comment]
Does this seem like a good candidate for async resolution? I think so. Proposal is:

- Keep `start`/`end` terms
- Return `null` when missing
- Don't include parenthesis

**Comment by bramus** [comment]
SGTM
